### DISCUSSION TITLE
Best way to create my own Charge

### DISCUSSION THREAD

**Discussion by jaahoo** [pay-rails/pay#682] [Q&A] [answered]
Hi guys,
I want to create my own Charge to Pay::Charge. It should have columns product_id, pay_charge_id, user_id, amount and status.
What will be the best way to do this?
1. Create it in controller after user.payment_processor.charge method.
pros: I can get all the info user_id, product_id, pay_charge_id etc.

2. Create it in "stripe.charge.succeeded" webhook response
I think this should be cleaner but I am not sure about possibility of multiple webhook call and the fact that I need to find Pay::Charge and there could be problem with that (my Charge is created before Pay::Charge)
I save product_id and user_id in metadata

3. Make concern and create my Charge in after_create callback. I dont try this one.

Thank you in advance for your advice and opinions

**Comment by excid3** [comment, marked as answer]
Adding an `after_create` or similar callback to the Pay::Charge model would work best because you can wait until Pay has handled the webhook and synced all the data.

We will probably expose some of these events in a Pay configuration in the future so you can define callbacks without having to use modules.

**Reply by jaahoo** [reply]
Thank you Chris for advice, it works like a charm.
